src/server/load_statistics.rs

```diff
@@ -0,0 +1,103 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Instant;
+
+use libc::{getpid, pid_t};
+
+use super::server::GRPC_THREAD_PREFIX;
+use util::metrics::{get_thread_ids, Stat};
+
+pub struct ThreadLoad {
+    term: AtomicUsize,
+    load: AtomicUsize,
+    threshold: usize,
+}
+
+impl ThreadLoad {
+    pub fn with_threshold(threshold: usize) -> Self {
+        ThreadLoad {
+            term: AtomicUsize::new(0),
+            load: AtomicUsize::new(0),
+            threshold,
+        }
+    }
+
+    #[allow(dead_code)]
```

Comment:
Can we do tests on these functions instead of allowing `dead_code`?

Reply:
Allowing `dead_code` is still needed even if we add tests for these functions because test cases are under `#[cfg(test)]`.